pytrip/utils/spc2pdf.py

```diff
@@ -0,0 +1,141 @@
+from pytrip import spc
+
+
+def main(args=sys.argv[1:]):
+
+    # parse arguments
+    parser = argparse.ArgumentParser()
+    parser.add_argument("spc_file", help="location of input SPC file", type=argparse.FileType('r'))
+    parser.add_argument("pdf_file", help="location of output PDF file", type=argparse.FileType('w'))
+    parser.add_argument("-v", "--verbosity", action='count', help="increase output verbosity", default=0)
+    parser.add_argument('-V', '--version', action='version', version=pt.__version__)
+    parsed_args = parser.parse_args(args)
+
+    if parsed_args.verbosity == 1:
+        logging.basicConfig(level=logging.INFO)
+    elif parsed_args.verbosity > 1:
+        logging.basicConfig(level=logging.DEBUG)
+    else:
+        logging.basicConfig()
+
+    # read SPC file
+    spc_object = spc.SPC(parsed_args.spc_file.name)
+    spc_object.read_spc()
+
+    # get total number of entries in SPC object, needed to allocate memory for numpy array
+    n = 0
+    for db in spc_object.data:
+        for sb in db.species:
+            n += int(sb.ne)
+    logging.info("Total number of entries in SPC file {}".format(n))
+
+    # allocate empty numpy array to store SPC data
+    data = np.recarray(
+        shape=(int(n), ),
+        dtype=[('depth', np.float64), ('z', np.int8), ('energy', np.float64), ('fluence', np.float64)])
+    logging.debug("Temporary numpy array to store SPC data has shape {}".format(data.shape))
+
+    # fill numpy array with spectral data, assuming that no references are used to store bin centers
+    n = 0
+    for db in spc_object.data:
+        for sb in db.species:
+            data.depth[n:n + int(sb.ne)] = db.depth
+            data.z[n:n + int(sb.ne)] = sb.z
+            data.energy[n:n + int(sb.ne)] = 0.5 * (sb.ebindata[:-1] + sb.ebindata[1:])  # bin centers
+            data.fluence[n:n + int(sb.ne)] = sb.histdata
+            n += int(sb.ne)
+
+    logging.debug("Temporary numpy array filled with data, size {}".format(data.size))
+
+    # find particle species, depth steps and energy bin centers
+    z_uniq = np.unique(data.z)
+    logging.info("Particle species Z: {}".format(z_uniq))
+
+    depth_uniq = np.unique(data.depth)
+    logging.info("{} of depth steps from {} to {}".format(len(depth_uniq), depth_uniq.min(), depth_uniq.max()))
+
+    energy_uniq = np.unique(data.energy)
+    logging.info("{} of depth energy bins from {} to {}".format(len(energy_uniq), energy_uniq.min(), energy_uniq.max()))
+
+    # save multiple page PDF with plots
+    with PdfPages(parsed_args.pdf_file.name) as pdf:
+        logging.debug("Saving PDF file {}".format(parsed_args.pdf_file.name))
+        plt.rc('text', usetex=False)
+
+        # first a summary plot of fluence vs depth, log scale on fluence (Y), several series corresponding to Z species
+        fig, ax = plt.subplots()
+        ax.set_title("")
+        ax.set_xlabel("Depth [cm]")
+        ax.set_ylabel("Fluence [a.u.]")
+        ax.set_yscale('log')
```

Comment:
I will also consider plotting fluence on linear scale by default and give the user a chance to change to scale to the logarithmic. @nbassler what do you think ?